diff -Nru plymouth-0.8.9/src/plugins/splash/script/plugin.c plymouth-0.8.9.new/src/plugins/splash/script/plugin.c
--- plymouth-0.8.9/src/plugins/splash/script/plugin.c	2013-09-13 16:46:14.630733637 +0800
+++ plymouth-0.8.9.new/src/plugins/splash/script/plugin.c	2013-09-13 17:35:35.127515128 +0800
@@ -40,6 +40,8 @@
 #include <values.h>
 #include <unistd.h>
 #include <wchar.h>
+#include <regex.h>
+#include <string.h>
 
 #include "ply-boot-splash-plugin.h"
 #include "ply-buffer.h"
@@ -516,6 +518,77 @@
   unpause_displays (plugin);
 }
 
+void clear_escape(char * src, size_t src_len, char * dst, size_t dst_len)
+{  
+    regmatch_t pmatch[2];
+    regex_t reg;
+    size_t p_len = 0, q_len = 0;
+    char * p, * q;
+
+    p = src;
+    q = dst;
+
+    if(regcomp(&reg,"^((\x1b\\[[0-9;]*m)|(\x1b\\[K))", REG_EXTENDED) != 0) {
+        return;
+    }
+    
+    while(p_len < src_len && q_len < dst_len) {
+        if(*p == '\x2e' || *p == '\r' || *p == '\n') {
+            p++;
+            p_len ++;
+        } else {
+            if(regexec(&reg, p, 2, pmatch, 0) == 0 && pmatch[1].rm_so == 0) {
+                p += pmatch[1].rm_eo;
+                p_len += pmatch[1].rm_eo;
+            } else {
+                *q = *p;
+                p++;
+                q++;
+                p_len ++;
+                q_len ++;
+            }
+        }
+    }
+
+    regfree(&reg);
+
+}
+
+
+static void
+boot_output (ply_boot_splash_plugin_t *plugin,
+                const char               *output, size_t size)
+{
+
+  char dst[8192];
+  char src[1024];
+  size_t length = size > sizeof(src) - 1 ? sizeof(src) - 1 : size;
+  int i;
+  char * token, * str, * saveptr;
+
+  ply_trace("boot_output: size %d output %s\n", size, output);
+
+  memcpy(src, output, length);
+  src[length] = 0;
+
+  for (i = 1, str = src; ; i++, str = NULL) {
+    token = strtok_r(str, "\r\n\x2e", &saveptr);
+    if(token == NULL)
+        break;
+    memset(dst, 0, sizeof(dst));
+    clear_escape(token, size, dst, sizeof(dst) - 1);
+    if(dst[0] == 0) {
+       continue;
+    }
+    pause_displays (plugin);
+    script_lib_plymouth_on_display_boot_message (plugin->script_state,
+                                              plugin->script_plymouth_lib,
+                                              dst);
+    unpause_displays (plugin);
+  }
+
+}
+
 static void
 hide_message (ply_boot_splash_plugin_t *plugin,
                  const char               *message)
@@ -549,6 +622,7 @@
     .display_question = display_question,
     .display_message = display_message,
     .hide_message = hide_message,
+    .on_boot_output = boot_output,
   };
 
   return &plugin_interface;
diff -Nru plymouth-0.8.9/src/plugins/splash/script/script-lib-plymouth.c plymouth-0.8.9.new/src/plugins/splash/script/script-lib-plymouth.c
--- plymouth-0.8.9/src/plugins/splash/script/script-lib-plymouth.c	2013-09-13 16:46:14.633733738 +0800
+++ plymouth-0.8.9.new/src/plugins/splash/script/script-lib-plymouth.c	2013-09-13 13:05:35.349274183 +0800
@@ -87,10 +87,17 @@
   data->script_display_message_func = script_obj_new_null ();
   data->script_hide_message_func = script_obj_new_null ();
   data->script_quit_func = script_obj_new_null ();
+  data->script_boot_message_func = script_obj_new_null ();
   data->mode = mode;
   
   script_obj_t *plymouth_hash = script_obj_hash_get_element (state->global, "Plymouth");
   script_add_native_function (plymouth_hash,
+                              "SetBootMessageFunction",
+                              plymouth_set_function,
+                              &data->script_boot_message_func,
+                              "function",
+                              NULL);
+  script_add_native_function (plymouth_hash,
                               "SetRefreshFunction",
                               plymouth_set_function,
                               &data->script_refresh_func,
@@ -299,6 +306,20 @@
   script_obj_unref (ret.object);
 }
 
+void script_lib_plymouth_on_display_boot_message (script_state_t             *state,
+                                             script_lib_plymouth_data_t *data,
+                                             const char                 *boot_message)
+{
+  script_obj_t *boot_message_obj = script_obj_new_string (boot_message);
+  script_return_t ret = script_execute_object (state,
+                                               data->script_boot_message_func,
+                                               NULL,
+                                               boot_message_obj,
+                                               NULL);
+  script_obj_unref (boot_message_obj);
+  script_obj_unref (ret.object);
+}
+
 void script_lib_plymouth_on_display_message (script_state_t             *state,
                                              script_lib_plymouth_data_t *data,
                                              const char                 *message)
diff -Nru plymouth-0.8.9/src/plugins/splash/script/script-lib-plymouth.h plymouth-0.8.9.new/src/plugins/splash/script/script-lib-plymouth.h
--- plymouth-0.8.9/src/plugins/splash/script/script-lib-plymouth.h	2013-09-13 16:46:14.606732823 +0800
+++ plymouth-0.8.9.new/src/plugins/splash/script/script-lib-plymouth.h	2013-09-13 12:50:30.729002948 +0800
@@ -39,6 +39,7 @@
   script_obj_t           *script_display_message_func;
   script_obj_t           *script_hide_message_func;
   script_obj_t           *script_quit_func;
+  script_obj_t           *script_boot_message_func;
   ply_boot_splash_mode_t  mode;
 } script_lib_plymouth_data_t;
 
@@ -79,4 +80,7 @@
 void script_lib_plymouth_on_quit (script_state_t             *state,
                                   script_lib_plymouth_data_t *data);
 
+void script_lib_plymouth_on_display_boot_message (script_state_t             *state,
+                                             script_lib_plymouth_data_t *data,
+                                             const char                 *boot_message);
 #endif /* SCRIPT_LIB_PLYMOUTH_H */
diff -Nru plymouth-0.8.9/src/plugins/splash/script/script-lib-plymouth.script plymouth-0.8.9.new/src/plugins/splash/script/script-lib-plymouth.script
--- plymouth-0.8.9/src/plugins/splash/script/script-lib-plymouth.script	2013-09-13 16:46:14.608732890 +0800
+++ plymouth-0.8.9.new/src/plugins/splash/script/script-lib-plymouth.script	2013-09-13 12:59:12.333455755 +0800
@@ -1,6 +1,6 @@
 #------------------------- Compatability Functions -------------------------
 Plymouth.SetMessageFunction = Plymouth.SetDisplayMessageFunction;
-
+PlymouthSetBootMessageFunction = Plymouth.SetBootMessageFunction;
 PlymouthSetRefreshFunction = Plymouth.SetRefreshFunction;
 PlymouthSetBootProgressFunction = Plymouth.SetBootProgressFunction;
 PlymouthSetRootMountedFunction = Plymouth.SetRootMountedFunction;
