--- /lib/dracut/modules.d/90dmsquash-live/dmsquash-live-root.sh	2016-07-08 18:21:21.000000000 +0800
+++ /lib/dracut/modules.d/90dmsquash-live.new/dmsquash-live-root.sh	2016-09-15 21:50:28.059346179 +0800
@@ -117,6 +117,13 @@
                 dd if=/dev/zero of=$OVERLAY_LOOPDEV bs=64k count=1 conv=fsync 2>/dev/null
             fi
             setup="yes"
+        elif [ ! -f /run/initramfs/overlayfs$pathspec ]; then
+            warn "Can not found persistent overlay, create one with size=$overlay_size MB"
+            mkdir -p /run/initramfs/overlayfs$pathspec
+            rm -rf /run/initramfs/overlayfs$pathspec
+            dd if=/dev/null of=/run/initramfs/overlayfs$pathspec bs=1024 count=1 seek=$((overlay_size * 1024)) 2> /dev/null &&
+            losetup $OVERLAY_LOOPDEV /run/initramfs/overlayfs$pathspec &&
+	    setup="yes"
         fi
         umount -l /run/initramfs/overlayfs || :
     fi
