diff -Nur 90dmsquash-live.ORG/dmsquash-live-root.sh 90dmsquash-live/dmsquash-live-root.sh
--- 90dmsquash-live.ORG/dmsquash-live-root.sh	2015-07-13 11:10:26.814251810 +0800
+++ 90dmsquash-live/dmsquash-live-root.sh	2015-07-13 11:08:12.048983251 +0800
@@ -117,6 +117,13 @@
                 dd if=/dev/zero of=$OVERLAY_LOOPDEV bs=64k count=1 conv=fsync 2>/dev/null
             fi
             setup="yes"
+        elif [ ! -f /run/initramfs/overlayfs$pathspec ]; then
+           warn "Can not found persistent overlay, create one with size=$overlay_size MB"
+           mkdir -p /run/initramfs/overlayfs$pathspec
+           rm -rf /run/initramfs/overlayfs$pathspec
+           dd if=/dev/null of=/run/initramfs/overlayfs$pathspec bs=1024 count=1 seek=$((overlay_size * 1024)) 2> /dev/null &&
+           losetup $OVERLAY_LOOPDEV /run/initramfs/overlayfs$pathspec &&
+	   setup="yes"
         fi
         umount -l /run/initramfs/overlayfs || :
     fi
@@ -259,6 +266,12 @@
 if [ -b "$OSMIN_LOOPDEV" ]; then
     # set up the devicemapper snapshot device, which will merge
     # the normal live fs image, and the delta, into a minimzied fs image
+    echo "0 $( blockdev --getsz $BASE_LOOPDEV ) snapshot $BASE_LOOPDEV $OSMIN_LOOPDEV p 8" | dmsetup create live-osimg-min
+    e2label /dev/mapper/live-osimg-min MINIME-LIVE
+    sync
+    dmsetup remove --retry live-osimg-min
+    sync
+    dmsetup remove --retry live-osimg-min
     echo "0 $( blockdev --getsz $BASE_LOOPDEV ) snapshot $BASE_LOOPDEV $OSMIN_LOOPDEV p 8" | dmsetup create --readonly live-osimg-min
 fi
 
@@ -273,3 +286,4 @@
 need_shutdown
 
 exit 0
+
diff -Nur 90dmsquash-live.ORG/module-setup.sh 90dmsquash-live/module-setup.sh
--- 90dmsquash-live.ORG/module-setup.sh	2015-07-13 11:10:26.815251931 +0800
+++ 90dmsquash-live/module-setup.sh	2015-07-13 11:08:50.268207468 +0800
@@ -22,7 +22,7 @@
 
 # called by dracut
 install() {
-    inst_multiple umount dmsetup blkid dd losetup grep blockdev
+    inst_multiple umount dmsetup blkid dd losetup grep blockdev e2label md5sum
     inst_multiple -o checkisomd5
     inst_hook cmdline 30 "$moddir/parse-dmsquash-live.sh"
     inst_hook cmdline 31 "$moddir/parse-iso-scan.sh"
